Firebase/Auth/FirebaseAuth.podspec

```diff
@@ -20,37 +20,60 @@ Simplify your iOS development, grow your user base, and monetize more effectivel
   s.social_media_url = 'https://twitter.com/Firebase'
   s.ios.deployment_target = '7.0'
+  s.osx.deployment_target = '10.10'
+
+  eitherSource = lambda { |paths|
+    Array(paths).map { |path| ['Firebase/Auth/Source/' + path, 'Source/' + path] }.flatten
+  }
+
+  s.source_files = eitherSource[[
+    'FirebaseAuth.h',
+    'FirebaseAuthVersion.h',
+    'FIRAdditionalUserInfo.h',
+    'FIRAuth.h',
+    'FIRAuthAPNSTokenType.h',
+    'FIRAuthCredential.h',
+    'FIRAuthDataResult.h',
+    'FIRAuthErrors.h',
+    'FIRAuthSwiftNameSupport.h',
+    'AuthProviders/EmailPassword/FIREmailAuthProvider.h',
+    'AuthProviders/Facebook/FIRFacebookAuthProvider.h',
+    'AuthProviders/GitHub/FIRGitHubAuthProvider.h',
+    'AuthProviders/Google/FIRGoogleAuthProvider.h',
+    'AuthProviders/OAuth/FIROAuthProvider.h',
+    'AuthProviders/Phone/FIRPhoneAuthCredential.h',
+    'AuthProviders/Phone/FIRPhoneAuthProvider.h',
+    'AuthProviders/Twitter/FIRTwitterAuthProvider.h',
+    'FIRUser.h',
+    'FIRUserInfo.h'
+  ]]
+
+  # Necessary hack to appease header visibility while as a direct OR transitive/internal dependency
+  s.subspec 'Internal' do |ss|
+    ss.source_files = eitherSource['**/*.[mh]']
+    ss.private_header_files = eitherSource['**/*.h']
+
+    ss.osx.exclude_files = eitherSource[[
+      'FIRAuthAppDelegateProxy.[mh]',
+      'FIRAuthNotificationManager.[mh]',
+      'FIRAuthAPNSTokenManager.[mh]',
+      'FIRAuthAPNSTokenType.[mh]',
+      'FIRAuthAPNSToken.[mh]',
+      'AuthProviders/Phone/FIRPhoneAuthProvider.[mh]'
+    ]]
+  end
 
-  s.source_files = '**/*.[mh]'
-  s.public_header_files =
-    'Source/FirebaseAuth.h',
-    'Source/FirebaseAuthVersion.h',
-    'Source/FIRAdditionalUserInfo.h',
-    'Source/FIRAuth.h',
-    'Source/FIRAuthAPNSTokenType.h',
-    'Source/FIRAuthCredential.h',
-    'Source/FIRAuthDataResult.h',
-    'Source/FIRAuthErrors.h',
-    'Source/FIRAuthSwiftNameSupport.h',
-    'Source/AuthProviders/EmailPassword/FIREmailAuthProvider.h',
-    'Source/AuthProviders/Facebook/FIRFacebookAuthProvider.h',
-    'Source/AuthProviders/GitHub/FIRGitHubAuthProvider.h',
-    'Source/AuthProviders/Google/FIRGoogleAuthProvider.h',
-    'Source/AuthProviders/OAuth/FIROAuthProvider.h',
-    'Source/AuthProviders/Phone/FIRPhoneAuthCredential.h',
-    'Source/AuthProviders/Phone/FIRPhoneAuthProvider.h',
-    'Source/AuthProviders/Twitter/FIRTwitterAuthProvider.h',
-    'Source/FIRUser.h',
-    'Source/FIRUserInfo.h'
   s.preserve_paths =
-    'README.md',
-    'CHANGELOG.md'
+    'Firebase/Auth/README.md', 'README.md',
+    'Firebase/Auth/CHANGELOG.md', 'CHANGELOG.md'
+
   s.xcconfig = { 'GCC_PREPROCESSOR_DEFINITIONS' =>
     '$(inherited) ' + 'FIRAuth_VERSION=' + s.version.to_s +
     ' FIRAuth_MINOR_VERSION=' + s.version.to_s.split(".")[0] + "." + s.version.to_s.split(".")[1]
   }
+
   s.framework = 'Security'
-#  s.dependency 'FirebaseDev/Core'
+  s.dependency 'FirebaseCore', '~> 4.0.1'
```

Comment:
This won't be publishable as long as the release version of FirebaseCore has closed source dependencies -
 https://github.com/CocoaPods/Specs/blob/master/Specs/8/b/d/FirebaseCore/4.0.0/FirebaseCore.podspec.json